src/R/Editor/Impl/Data/WorkspaceVariableProvider.cs

```diff
@@ -111,13 +115,42 @@ internal sealed class WorkspaceVariableProvider : RSessionChangeWatcher, IVariab
                                     .ToArray();
                         }
                     }
+                } else {
```

Comment:
Given this branch, is the one above still needed? It looks like evaluating the head of the expression will always give the right answer, whether the expression involves a package or not.